### ISSUE TITLE
Additional button in header to filter through data before searching

### ISSUE THREAD

**Issue by StephenStemSoftware** [eakoriakin/ionic-selectable#129]
Hello, 

I love your plugin and have been using it for a while now on a few of my apps.
I am wondering if it would be possible to add a button on the header to filter through the data. For example, I use GPS locations for the values and I want the user to be able to press a button and only the values that are within a certain distance will populate.

 I looked at the Add, Update, and Delete. However, it seems those are manual inputs. I would just like the button to call a function to dynamically manipulate the array and display to the user the values if any.

Just a suggestion here, for I am using a similar function my app that achieves this. It would just be awesome if it were possible in your plugin. 

Cheers,
Stephen

**Comment by kodeine** [comment]
i was looking for the same thing. i think we can use something like this

```
<ng-template selectSearchableTitleTemplate>
        {{ title || 'Pick A Location' }}
<ion-button>......
    </ng-template>
```

I will test this and see if it works.

**Comment by eakoriakin** [comment]
Hey guys, it’s now possible to add additional buttons in the footer:
https://github.com/eakoriakin/ionic-selectable/wiki#footer-template

**Comment by StephenStemSoftware** [comment]
Thank you @eakoriakin  for your continual efforts to make this a worthwhile plugin. I was curious, with these buttons, is it possible to dynamically change what data is on showing? 

For example, Let's say the values have gps coordinates and the user wanted to press a button to grab the nearest locations from a certain distance away. The array would then be filtered and show the nearest values dynamically on the screen?

I have an app that uses this feature, but I am currently having to show and hide two different form fields to achieve this.

Thank You,
Stephen

**Comment by eakoriakin** [comment]
Hey Stephen, sounds like you just need to add a custom button, say Nearest, to the footer that when pressed would filter items array.

**Comment by StephenStemSoftware** [comment]
I apologize I am still fairly new to the Angular and Typescript world. So I have my buttons I need set, but when I update the array nothing shows unless i close out of the plugin and go back into it. Then it will show the updated values correctly.

**Comment by eakoriakin** [comment]
Here is how to change footer buttons:

Sample
https://github.com/eakoriakin/ionic-selectable/wiki#footer-template

And a demo
https://stackblitz.com/edit/ionic-selectable-footer-template?file=app/pages/home/home.html

**Comment by StephenStemSoftware** [comment]
Okay I was able to get the buttons and data set, but for your plugin am I needing to update the variable? Or with me resigning it, is that what is causing the issue? 

I created a question on stack overflow to help assist you on what I am trying to achieve
[link](https://stackoverflow.com/questions/55400670/filter-though-object-array-based-on-distance-between-gps-locationsjavascript)

The code is updating the location, but not updating in the plugin. 
I apologize if this is out of the scope, but I've been trying to get this to work and no luck
